registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java

```diff
@@ -56,5 +102,150 @@ public void saveAnonymousProfile(String regId, String processStage, String profi
 					"AnonymousProfileServiceImpl::saveAnonymousProfile()::exit");
 
 	}
+	
+	@Override
+	public String buildJsonStringFromPacketInfo(BiometricRecord biometricRecord, Map<String, String> fieldMap,
+			Map<String, String> metaInfoMap, String statusCode, String processStage)
+			throws JSONException, ApisResourceAccessException, PacketManagerException, IOException {
+
+		AnonymousProfileDTO anonymousProfileDTO = new AnonymousProfileDTO();
+		anonymousProfileDTO.setProcessName(
+				getFieldValueFromMetaInfo(metaInfoMap, JsonConstant.METADATA, JsonConstant.REGISTRATIONTYPE));
+		anonymousProfileDTO.setProcessStage(processStage);
+		anonymousProfileDTO.setStatus(statusCode);
+		String date = metaInfoMap.get(JsonConstant.CREATIONDATE);
+		anonymousProfileDTO.setDate(DateUtils.getTimeFromDate(
+				DateUtils.parseToDate(date, ANONYMOUS_PROFILE_DATETIME_PATTERN), ANONYMOUS_PROFILE_DATETIME_PATTERN));
+
+		String dateArr[] = fieldMap.get("dateOfBirth").split("/");
+		anonymousProfileDTO.setYearOfBirth(dateArr[0]);
+		anonymousProfileDTO.setGender(getLanguageBasedValueForSimpleType(fieldMap.get(MappingJsonConstants.GENDER)));
+
+		String zone = getLanguageBasedValueForSimpleType(fieldMap.get("zone"));

```

Comment:
this is not correct. you should read address field from registration-processor-identity.json
the values will be comma separated location hierarchy. take last 2 element of the location hierarchy.

Reply:
corrected